engines/private/private.cpp

```diff
@@ -0,0 +1,1081 @@
+                rs = cs;
+                ns = e.nextSetting;
+            }
+        }
+    }
+    if (ns != NULL) {
+        //debug("Exit selected %s", ns->c_str());
+        _nextSetting = ns;
+    }
+}
+
+void PrivateEngine::selectMask(Common::Point mousePos) {
+    Common::String *ns = NULL;
+    MaskInfo m;
+    for (MaskList::iterator it = _masks.begin(); it != _masks.end(); ++it) {
+        m = *it;
+        //debug("Testing mask %s", m.nextSetting->c_str());
+        if (inMask(m.surf, mousePos)) {
+            //debug("Inside!");
+            if (m.nextSetting != NULL) { // TODO: check this
+                //debug("Found Mask %s", m.nextSetting->c_str());
+                ns = m.nextSetting;
+            }
+
+            if (m.flag1 != NULL) { // TODO: check this
+                setSymbol(m.flag1, 1);
+                // an item was taken
+                if (_toTake) {
+                    playSound(*getTakeSound(), 1, false, false);
+                    _toTake = false;
+                }
+            }
+
+            if (m.flag2 != NULL) {
+                setSymbol(m.flag2, 1);
+            }
+            break;
+        }
+    }
+    if (ns != NULL) {
+        //debug("Mask selected %s", ns->c_str());
+        _nextSetting = ns;
+    }
+}
+
+void PrivateEngine::selectAMRadioArea(Common::Point mousePos) {
+    if (_AMRadioArea == NULL)
+        return;
+
+    if (_AMRadio.empty())
+        return;
+
+    if (inMask(_AMRadioArea->surf, mousePos)) {
+        Common::String sound = *_infaceRadioPath + "comm_/" + _AMRadio.back() + ".wav";
+        playSound(sound.c_str(), 1, false, false);
+        _AMRadio.pop_back();
+    }
+}
+
+void PrivateEngine::selectPoliceRadioArea(Common::Point mousePos) {
+    if (_policeRadioArea == NULL)
+        return;
+
+    if (_policeRadio.empty())
+        return;
+
+    if (inMask(_policeRadioArea->surf, mousePos)) {
+        Common::String sound = *_infaceRadioPath + "police/" + _policeRadio.back() + ".wav";
+        playSound(sound.c_str(), 1, false, false);
+        _policeRadio.pop_back();
+    }
+}
+
+void PrivateEngine::checkPhoneCall() {
+    if (_phoneArea == NULL)
+        return;
+
+    if (_phone.empty())
+        return;
+
+    if (!_mixer->isSoundHandleActive(_fgSoundHandle))
+        playSound(*_phonePrefix + *_phoneCallSound, 1, false, false);
+}
+
+void PrivateEngine::selectPhoneArea(Common::Point mousePos) {
+    if (_phoneArea == NULL)
+        return;
+
+    if (_phone.empty())
+        return;
+
+    if (inMask(_phoneArea->surf, mousePos)) {
+        PhoneInfo i = _phone.back();
+        Common::String sound(*i.sound);
+        setSymbol(i.flag, i.val);
+        sound = *_phonePrefix + sound + ".wav";
+        playSound(sound.c_str(), 1, true, false);
+        _phone.pop_back();
+    }
+}
+
+void PrivateEngine::loadDossier() {
+    int x = 40;
+    int y = 30;
+    int i = _dossierSuspect;
+    int j = _dossierPage;
```

Comment:
both `i` and `j` could be dropped and `_dossierSuspect` and `_dossierPage` used in-place, for better code readability.

Reply:
Done.